core/src/datum.rs

```diff
@@ -320,6 +320,27 @@ impl TryInto<f64> for bool {
     }
 }
 
+macro_rules! cast_to_and_from_bool {
+    ($t: ty) => {
+        impl TryInto<bool> for $t {
+            fn try_into(&self) -> TractResult<bool> {
+                Ok(*self == 0)
```

Comment:
The int -> bool cast is not well defined, but according to https://github.com/onnx/onnx/issues/2285 , it seems that truncating the leading bits is the right approach. Suggestion:
```suggestion
                Ok((*self & 1) == 1)
```

Reply:
wow. I was more tired than I thought.